### PR TITLE
Add missing list view usages to data type references (#14485)

### PR DESCRIPTION
**Pull request by ustadstar**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/14485

Please see the issue for the description of the bug including the steps to reproduce. 

My changes do show the content types which use a list view as a container under references on the info tab of a datatype. However my changes are not the ideal solution to the problem. A better solution for the future would be a change in the database schema to add a relation between the list view (the isContainer field on contentType is now a boolean, without any relation). 

My solution finds the relation between the datatype and list views as a container as follows:

- We check if the current data type is a list view, if it is we continue 
- Next we check if the name of the datatype matches the custom list view name.  The logic behind this is the same as in: ListViewContentAppFactory.cs. If it is not a custom list view we continue
- Next we check if the datatype name matches the default content type list view name. if not we return an empty array.

To test this PR please follow the steps below.

**Test with a default list view**
1. Navigate to settings
2. Choose a document type
3. Open the list view tab
4. Enable list view and save
5. Navigate to settings -> data types -> List View - Content
6.  Open the info panel
7.  Check if List View - Content is listed under references

image

**Test with a custom list view**
1. Navigate to settings
2. Create a new document type, e.g. "Content Page"
3. Open the list view tab
4. Enable list view and choose "Create a custom list view"
5. Save the new document type
6. Navigate to settings -> data types -> List View - contentPage
7.  Open the info panel
8.  Check if Content Page is listed under references

image

### CODE REVIEW

File: src/Umbraco.Core/Persistence/Repositories/IDataTypeRepository.cs [hunk 1]
```diff
@@ -14,4 +14,13 @@ public interface IDataTypeRepository : IReadWriteQueryRepository<int, IDataType>
...
     IReadOnlyDictionary<Udi, IEnumerable<string>> FindUsages(int id);
+
+    /// <summary>
+    ///     Returns a dictionary of content type <see cref="Udi" />s and the data type (List view) aliases that use a
+    ///     <see cref="IDataType" />
+    /// </summary>
+    /// <param name="id"></param>
+    /// <returns></returns>
+
+    IReadOnlyDictionary<Udi, IEnumerable<string>> FindListViewUsages(int id);
```
Comment (nul800sebastiaan) [inline src/Umbraco.Core/Persistence/Repositories/IDataTypeRepository.cs:25]: This is a breaking change, anybody already implementing this interface will get a build error and plugins that haven't been updated will make the whole site error out.

Luckily there's an easy way to work around this and that is to introduce a default implementation, an example can be seen here: https://github.com/umbraco/Umbraco-CMS/pull/11867#issuecomment-1178869417

**Review by nul800sebastiaan — CHANGES_REQUESTED** [review]
Thanks @ustadstar - we'll have a better look soon, but to begin with I've left you a comment to update the interface so it's not a breaking change.

File: src/Umbraco.Core/Services/IDataTypeService.cs [hunk 2]
```diff
@@ -14,6 +14,7 @@ public interface IDataTypeService : IService
     /// <param name="id"></param>
     /// <returns></returns>
     IReadOnlyDictionary<Udi, IEnumerable<string>> GetReferences(int id);
+    IReadOnlyDictionary<Udi, IEnumerable<string>> GetListViewReferences(int id);
```
Comment (nul800sebastiaan) [inline src/Umbraco.Core/Services/IDataTypeService.cs:17]: Same as the other interface, this is also a breaking change, which can be fixed in the same way.

**Comment by ustadstar** [comment]
@nul800sebastiaan I have added a default implementation for both methods in the interfaces.

**Comment by ustadstar** [comment]
Hi @nul800sebastiaan, did you have time to review the changes?

**Comment by georgebid** [comment]
Hey, @ustadstar! I thought I'd jump in here and review this PR 😄 Thanks a lot for making those change requests, I have tested this PR and everything looks good to me, therefore I have merged this one in! Thanks for fixing #14485 🚀 You'll see this in the next release.

**Comment by emmagarland** [comment]
Great stuff all!

**Pull request merged**